CAIPs/caip-25.md

```diff
@@ -32,25 +32,55 @@ application through a provider connecting to a wallet.
-reasons, it can simply generate a new session identifier to signal this event to
-the calling provider.
-
-The application interfaces with a provider to populate a session with a base
-state describing authorized chains, methods, event, and accounts.  This
-negotation takes place by sending the application's REQUIRED and REQUESTED
-properties of the session. If any requirements are not met, a failure response
-expressive of one or more specific failure states will be sent (see below).
+updated, extended, closed, etc by successive calls and events. The exact
+parameters and assumptions of that session abstraction are defined in
+[CAIP-171][], but note that a string identifier referring to it is absent from
+the initial call (if authorization is granted) and present in both the initial
+response and all future responses.
+
+Given the session model of [CAIP-171][], this interface outlines the
+authorization of a provider to handle a set of interfaces grouped into
+namespaces, as well as to interact with a session abstraction used by both
+caller and respondent to manage the authorization over time. The
+`sessionIdentifier` defined in [CAIP-171][] enables this mutual management and
+alignment across calls that are idempotent if identical. If a respondent (e.g. a
+wallet) needs to initiate a new session, whether due to user input, security
+policy, or session expiry reasons, it can simply generate a new session
+identifier to signal this event to the calling provider; if a caller needs to
+initiate a new session, it can do so by sending a new request without
+`sessionIdentifier`. In such cases, a respondent (e.g. wallet) may choose to
+explicitly close all sessions upon generation of a new one from the same origin,
+or leave it to time-out; maintaining concurrent sessions is discouraged (see
+Security Considerations).
+
+In the initial call, the application interfaces with a provider to populate a
+session with a base state describing authorized chains, methods, event, and
+accounts.  This negotation takes place by sending the application's REQUIRED and
+REQUESTED authorizations of the session, grouped into objects scoping those
+authorizations which in turn are grouped into two top-level arrays (named
+`requiredScopes` and `optionalScopes` respectively).  These two arrays are not
+mutually exclusive (i.e., additional properties of a required scope may be
+requested under the same keyed scope object key in the requested array). Note
+that scopes can be keyed to an entire [CAIP-104][] "namespace", meaning
+applicable to *any* current or future [CAIP-2][] chainID within that namespace,
+or keyed to a specific [CAIP-2][] within that namespace.
+
+If any properties in the required scope(s) are not authorized by the
+respondent (e.g. wallet), a failure response expressive of one or more specific
+failure states will be sent (see [#### failure states](#failure-states) below),
+with the exception of user denying consent. For privacy reasons, an `undefined`
```

Comment:
This needs to be updated in line with @pedrouid's comment from last week, not sure if this is a point of contention or not, but I agree with his point.

Reply:
Sure-- will finesse the language in a new commit

Reply:
[Voilà](https://github.com/ChainAgnostic/CAIPs/pull/206/commits/c80ea5662cc290577615cf6bed36e4b7ecf63917)

Reply:
Perfect! Just wondering, how come the error code # change?